controllers/openstackclient_controller.go

```diff
@@ -96,7 +96,10 @@ func (r *OpenStackClientReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 	//
 	// initialize condition
 	//
-	cond := &ospdirectorv1beta1.Condition{}
+	cond := instance.Status.Conditions.GetCurrentCondition()
+	if cond == nil {
+		cond = &ospdirectorv1beta1.Condition{}
+	}
```

Comment:
how about to create a instance.Status.Conditions.InitCondition() which does these tasks? I mean GetCurrentCondition() and if nil returns &ospdirectorv1beta1.Condition{}

Reply:
Good idea

Reply:
Done